guard-lambda/src/main.rs

```diff
@@ -1,44 +1,47 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-use std::error::Error;
-
 use cfn_guard;
-use lambda_runtime::{error::HandlerError, lambda, Context};
-use log::{self, info};
+use lambda_runtime::{handler_fn, Context, Error};
+use log::{self, LevelFilter, info};
 use serde_derive::{Deserialize, Serialize};
-use simple_logger;
+use simple_logger::SimpleLogger;
 
 #[derive(Deserialize, Debug)]
 struct CustomEvent {
     #[serde(rename = "data")]
     data: String,
     #[serde(rename = "rules")]
-    rules: String,
+    rules: Vec<String>,
 }
 
 #[derive(Serialize)]
 struct CustomOutput {
-    message: String,
+    message: Vec<serde_json::Value>,
 }
 
-fn main() -> Result<(), Box<dyn Error>> {
-    simple_logger::init_with_level(log::Level::Info)?;
-    lambda!(call_cfn_guard);
+#[tokio::main]
+async fn main() -> Result<(), Error> {
 
+    SimpleLogger::new().with_level(LevelFilter::Info).init().unwrap();
+    let func = handler_fn(call_cfn_guard);
+    lambda_runtime::run(func).await?;
     Ok(())
 }
 
-fn call_cfn_guard(e: CustomEvent, _c: Context) -> Result<CustomOutput, HandlerError> {
-    info!("Template is [{}]", &e.data);
-    info!("Rule Set is [{}]", &e.rules);
-    let result = match cfn_guard::run_checks(&e.data, &e.rules)
-    {
-        Ok(t) => t,
-        Err(e) => (e.to_string()),
-    };
-
+pub(crate) async fn call_cfn_guard(e: CustomEvent, _c: Context) -> Result<CustomOutput, Error> {
```

Comment:
With #154 we have support for output reporters that can provide lesser detail but sufficient for 80% of the use cases. Would you want to expose these options via lambda invocation as well? Take the look at example on that PR esp. with `--show-summary none --output-format yaml`

Reply:
Yes, I can make those changes. Thanks!

Reply:
Let us do that as a separate PR

Reply:
Sounds good.